convert_components_to_json.py

```diff
@@ -156,7 +156,65 @@ def convert_components_to_json():
                             component_info["channels"] = 8
                         else:
                             component_info["channels"] = 4
-                
+
+                    # Special handling for GPS over I2C
+                    if component_data.get("isGps", False):
+                            print("**** GPS over I2C detected for component: " + component_dir)
+                            component_info["isGps"] = True
+                    if "commands_ubxes" in component_data:
+                        component_info["commands_ubxes"] = component_data["commands_ubxes"]
+                    if "commands_pmtks" in component_data:
+                        component_info["commands_pmtks"] = component_data["commands_pmtks"]
+
+                # Handle UART-specific properties
+                if category == "uart":
+                    # Required properties
+                    component_info["device_type"] = component_data.get("deviceType")
+                    component_info["deviceId"] = component_data.get("deviceId")
+                    # Specific device_type properties
+                    if component_info["device_type"] == "generic_input":
+                        # Parse generic input properties
+                        component_info["period"] = component_data["generic_input"].get("period", 30000)  # Default to 30s if not specified
+                        # Extract data types
+                        if "generic_input" in component_data and "sensor_types" in component_data["generic_input"]:
+                            for meas_type in component_data["generic_input"]["sensor_types"]:
+                                if isinstance(meas_type, dict) and "sensorType" in meas_type:
+                                    component_info["dataTypes"].append({
+                                        "displayName": meas_type["displayName"] if "displayName" in meas_type else meas_type["sensorType"],
+                                        "sensorType": map_datatypes_to_offline_types(meas_type["sensorType"]) if "sensorType" in meas_type else None
+                                    })
+                                else:
+                                    component_info["dataTypes"].append(map_datatypes_to_offline_types(meas_type))
+                    elif component_info["device_type"] == "gps":
+                        component_info["period"] = component_data.get("period", 30000)  # Default to 30s if not specified
+                        if "commands_ubxes" in component_data:
+                            component_info["commands_ubxes"] = component_data["commands_ubxes"]
+                        if "commands_pmtks" in component_data:
+                            component_info["commands_pmtks"] = component_data["commands_pmtks"]
```

Comment:
nested under component_info["gps"]["period"] etc
I think having the default unlisted has allowed this to slip through the net, maybe specifying it in the component json definitions works better. The validation schema suggests is required if the gps section exists. However by having devicetype=gps and no properties this has slipped through.

The PM25 beneath looks like it does it correctly